### PR TITLE
Combine unit and e2e tests into a single GHA workflow + add e2e test reporter

### PR DESCRIPTION
**Pull request by megahirt**
## Description

This is a follow up PR to #1278

A recent pairing session with @longrunningprocess led us to want to try and combine all tests into a single workflow for efficiency and simplicity.  Now that I know the status checks can still be separate for a single workflow, it makes sense to combine all tests into a single workflow and have separate reporters / checks.

### Type of Change

- engineering

## Testing on GHA

Please describe how to test and/or verify your changes. Provide instructions so we can reproduce.  Please also provide relevant test data as necessary.  These instructions will be used for QA testing below.

- [x] E2E tests and Unit tests run concurrently (separate jobs)
- [x] E2E tests are reported as separate status checks
- [x] E2E JUnit results are reported as a comment in the PR

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas

## qa.languageforge.org testing

Reviewers: add/replace your name below and check the box to sign-off/attest the feature works as expected on qa.languageforge.org

- [ ] Reviewer1 (YYYY-MM-DD HH:MM)
- [ ] Reviewer2 (YYYY-MM-DD HH:MM)

### CODE REVIEW

File: test/app/protractorConf.js [hunk 1]
```diff
@@ -28,34 +28,41 @@ exports.config = {
...
+      jasmine.getEnv().addReporter(junitReporter);
     }
+    var SpecReporter = require('jasmine-spec-reporter').SpecReporter;
+    jasmine.getEnv().addReporter(new SpecReporter({
+      spec: {
+        displayStacktrace: true
+      }
+    }));
+    /*
+    jasmine.getEnv().addReporter(new jasmineReporters.TerminalReporter({
```
Comment (longrunningprocess) [inline test/app/protractorConf.js:50]: any reason to hang onto this commented code anymore?

**Review by longrunningprocess — APPROVED** [review]
nice change!  Would you mind changing `unit-tests.yml` to `tests.yml` before you merge?

File: test/app/protractorConf.js (hunk 1 above)
Comment (megahirt) [reply test/app/protractorConf.js]: I deleted the TerminalReporter code - not needed, thanks.  The other function is potentially useful for debugging e2e tests

**Comment by megahirt** [comment]
> nice change! Would you mind changing `unit-tests.yml` to `tests.yml` before you merge?

Yes, I meant to do that.  Thanks for catching it.

**Pull request merged**